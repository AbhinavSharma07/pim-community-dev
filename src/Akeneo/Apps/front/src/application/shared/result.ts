interface Ok<T> {
    readonly ok: T;
}

interface Err<T> {
    readonly err: T;
}

export type Result<T, E> = Ok<T> | Err<E>;

export const ok = <T>(ok: T): Ok<T> => ({ok});

export const err = <T>(err: T): Err<T> => ({err});

export const isOk = <T, E>(result: Result<T, E>): result is Ok<T> => 'ok' in result;

export const isErr = <T, E>(result: Result<T, E>): result is Err<E> => 'err' in result;
