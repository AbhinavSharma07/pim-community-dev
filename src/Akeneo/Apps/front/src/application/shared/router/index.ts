import {BreadcrumbRouterLink} from './BreadcrumbRouterLink';
import {RouterContext} from './router-context';
import {Router as RouterInterface} from './router.interface';

export {RouterContext, RouterInterface, BreadcrumbRouterLink};
