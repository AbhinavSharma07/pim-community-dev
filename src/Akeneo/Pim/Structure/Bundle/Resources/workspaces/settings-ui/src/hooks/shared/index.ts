export * from './useDataGridState';
export * from './useMountedRef';
export * from './useSetPageTitle';
