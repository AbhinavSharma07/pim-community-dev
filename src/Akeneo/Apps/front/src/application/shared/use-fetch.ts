import {useState, useEffect} from 'react';
import {err, ok, Result} from './result';

const defaultInit: RequestInit = {
    credentials: 'include'
};

export const useFetch = <T, E>(input: RequestInfo, init?: RequestInit): Result<T | undefined, E> => {
    init = {...defaultInit, ...init};

    const [result, setResult] = useState<Result<T | undefined, E>>(ok(undefined));

    useEffect(() => {
        const doFetch = async () => {
            const response = await fetch(input, init);
            if (!response.ok) {
                setResult(err<E>(await response.json()));

                return;
            }
            setResult(ok<T>(await response.json()));
        };
        doFetch();
    }, [input, init]);

    return result;
};
