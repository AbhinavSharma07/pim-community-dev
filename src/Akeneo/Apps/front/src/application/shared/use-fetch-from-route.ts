import React from 'react';
import {RouterContext} from './router/router-context';
import {useFetch} from './use-fetch';

export const useFetchFromRoute = <T, E>(route: string, parameters?: object) => {
    const {generate} = React.useContext(RouterContext);

    return useFetch<T, E>(generate(route, parameters));
};
