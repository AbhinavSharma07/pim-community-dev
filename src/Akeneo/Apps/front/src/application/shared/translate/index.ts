import {Translate} from './Translate';
import {TranslateContext} from './translate-context';
import {Translate as TranslateInterface} from './translate.interface';

export {TranslateContext, Translate, TranslateInterface};
