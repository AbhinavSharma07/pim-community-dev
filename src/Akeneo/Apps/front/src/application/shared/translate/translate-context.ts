import * as React from 'react';
import {Translate} from './translate.interface';

export const TranslateContext = React.createContext<Translate>(id => id);
