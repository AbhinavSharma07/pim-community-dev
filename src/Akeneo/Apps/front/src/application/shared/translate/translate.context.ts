import * as React from 'react';
import {translate} from './translate.interface';

export const TranslateContext = React.createContext<translate>(id => id);
