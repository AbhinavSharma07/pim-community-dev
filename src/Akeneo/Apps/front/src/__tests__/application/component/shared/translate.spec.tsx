import * as React from 'react';
import { create } from 'react-test-renderer';
import { Translate } from '../../../../application/component/shared/translate';

describe('Translate component', () => {
    test('Matches the snapshot', () => {
        const translate = create(<Translate id='hello.world' />);
        expect(translate.toJSON()).toMatchSnapshot();
    });
});
