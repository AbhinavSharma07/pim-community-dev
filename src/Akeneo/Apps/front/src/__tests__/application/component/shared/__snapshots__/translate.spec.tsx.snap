// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Translate component Matches the snapshot 1`] = `"hello.world"`;
